crates/starknet-os/src/hints/tests.rs

```diff
@@ -278,4 +282,136 @@ pub mod tests {
+        insert_value_from_var_name(vars::ids::RANGE_CHECK96_PTR, rc96_segment, &mut vm, &ids_data, &ap_tracking)
+            .expect("insert_value_from_var_name");
+
+        let rc96_base = with_offset(rc96_segment, 0);
+        vm.insert_value(rc96_base, Felt252::THREE).expect("insert value at base");
+        for i in 1..rc96_segment.offset {
+            let address = with_offset(rc96_segment, i);
+            assert_eq!(vm.get_maybe(&address), None);
+        }
+
+        fill_holes_in_rc96_segment(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants)
+            .expect("fill_holes_in_rc96_segment failed");
+
+        // Make sure existing value isn't overwritten
+        assert_eq!(vm.get_maybe(&rc96_base), Some(Felt252::THREE.into()));
+
+        for i in 1..rc96_segment.offset {
+            let address = with_offset(rc96_segment, i);
+            assert_eq!(vm.get_maybe(&address), Some(Felt252::ZERO.into()));
+        }
+    }
+
+    #[test]
+    fn test_search_sorted_optimistic_with_zero_sized_elements() {
+        let mut vm = VirtualMachine::new(false);
+        vm.add_memory_segment();
+        vm.add_memory_segment();
+        vm.set_fp(2);
+
+        let mut exec_scopes = ExecutionScopes::new();
+        let ids_data = ids_data![ARRAY_PTR, ELM_SIZE];
+        let ap_tracking = ApTracking::default();
+        let constants = HashMap::new();
+
+        let array_ptr = vm.add_memory_segment();
+        insert_value_from_var_name(ARRAY_PTR, array_ptr, &mut vm, &ids_data, &ap_tracking).unwrap();
+        insert_value_from_var_name(ELM_SIZE, Felt252::ZERO, &mut vm, &ids_data, &ap_tracking).unwrap();
+
+        let result = search_sorted_optimistic(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants);
+        assert!(matches!(result, Err(HintError::AssertionFailed(_))), "{:?}", result);
+    }
+
+    #[test]
+    fn test_search_sorted_optimistic_with_too_many_elements() {
+        let mut vm = VirtualMachine::new(false);
+        vm.add_memory_segment();
+        vm.add_memory_segment();
+        vm.set_fp(3);
+
+        let mut exec_scopes = ExecutionScopes::new();
+        set_variable_in_root_exec_scope(&mut exec_scopes, vars::scopes::FIND_ELEMENT_MAX_SIZE, Some(2usize));
+        let ids_data = ids_data![ARRAY_PTR, ELM_SIZE, N_ELMS];
+        let ap_tracking = ApTracking::default();
+        let constants = HashMap::new();
+
+        let array_ptr = vm.add_memory_segment();
+        insert_value_from_var_name(ARRAY_PTR, array_ptr, &mut vm, &ids_data, &ap_tracking).unwrap();
+        insert_value_from_var_name(ELM_SIZE, Felt252::ONE, &mut vm, &ids_data, &ap_tracking).unwrap();
+        insert_value_from_var_name(N_ELMS, Felt252::THREE, &mut vm, &ids_data, &ap_tracking).unwrap();
+
+        let result = search_sorted_optimistic(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants);
+        assert!(matches!(result, Err(HintError::AssertionFailed(_))), "{:?}", result);
```

Comment:
The error message in `AssertionFailed` would be useful to have here for understanding what the test is expecting to fail